### PR TITLE
MovingAverage optimizer example 

### LINKED ISSUE #927: Add a tutorial for MovingAverage optimizer
We need a tutorial that shows how to use `MovingAverage` optimizer along with the custom callback for saving: `AverageModelCheckpoint`. Can alternatively use `StochasticAverage`. 

~~Currently blocked by #926~~

### PR DESCRIPTION
**Pull request by abhichou4**
Closes #927

### CODE REVIEW

**Comment by abhichou4** [comment]
Please let me know what more I can add to this and what I should change.

File: docs/tutorials/optimizers_movingaverage.ipynb [hunk 1]
```diff
@@ -0,0 +1,321 @@
+{
```
Comment (seanpmorgan) [inline docs/tutorials/optimizers_movingaverage.ipynb]: <p>Could you confirm that the <code>-t &lt;path&gt; is necessary</code>  or does it work without specifying that</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/addons/pull/1057/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (seanpmorgan) [inline docs/tutorials/optimizers_movingaverage.ipynb]: <p>Could you compare this against a more typical optimizer for comparison</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/addons/pull/1057/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

Comment (seanpmorgan) [inline docs/tutorials/optimizers_movingaverage.ipynb]: <p>Could you leave the cell results visible so it is easier to follow without having to run the notebook</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/addons/pull/1057/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

**Comment by Squadrick** [comment]
<p><code>ModelAverageCheckpoint</code>  should also be used for <code>StocasticAverage</code>  optimizer as well. Comparing <code>MovingAverage</code>  vs. <code>StocasticAverage</code>  vs. unwrapped optimizer for the same dataset training would be a pretty nice tutorial overall. </p>

File: docs/tutorials/optimizers_movingaverage.ipynb (hunk 1 above)
Comment (abhichou4) [reply docs/tutorials/optimizers_movingaverage.ipynb]: It works without that. Removing it.

**Comment by abhichou4** [comment]
This is a mess, apologies. Closing this and making another PR.

**Pull request closed without merging**